### DISCUSSION TITLE
Can quarto template use a local git directory?

### DISCUSSION THREAD

**Discussion by nucleic-acid** [quarto-dev/quarto-cli#7650] [Q&A]
### Description

I'd like to regularly use a quarto letter extension/template (in particular [this one](https://github.com/mavam/quarto-brief)) but with my personal address and contact information already included.

I don't want to upload this information in a fork of the template, but rather have quarto pull the template from a local clone on my machine of the repository. Like

`quarto use template /path/to/my/local/quarto-brief`

Trying this didn't work. Is there a way? I cannot find anything like it in the documentation.

**Comment by mcanouil** [comment]
See the documentation on distributing extensions: https://quarto.org/docs/extensions/creating.html#distribution

**Reply by bjornasm** [reply]
What I feel could be added to the [documentation](https://quarto.org/docs/journals/formats.html) is that you can do:

`quarto use template path/to/local/folder`


I also think it would be nice be a bit more specific in the workflow from creating, adding and using extensions and general. I would guess most would like to use and test the extension locally before putting it on an open git repo.

**Reply by mcanouil** [reply]
That's documented under "distributing":

- https://quarto.org/docs/extensions/distributing.html

Please avoid resurrecting 2-years old post which has possibly outdated answer.
In this case, the documentation has a link to the page above.

Note that the page you linked is about creating extension.

**Reply by bjornasm** [reply]
> That's documented under "distributing":
> 
> * https://quarto.org/docs/extensions/distributing.html
> 
> Please avoid resurrecting 2-years old post which has possibly outdated answer. In this case, the documentation has a link to the page above.
> 
> Note that the page you linked is about creating extension.

Sorry @mcanouil - I searched the problem and this popped up so I thought it would be ok to add it there. I believe testing/using the extension is relevant for anyone creating it, imo. Especially this edge case where you can use it "locally" without distributing it on github.

**Reply by mcanouil** [reply]
@bjornasm Feedbacks are all welcome. There is nothing to be sorry about.

Let me clarify, the distributing part can be accessed through the creating an extension and I personally don't believe duplicating the whole page will improve the experience.

> Especially this edge case where you can use it "locally" without distributing it on github.

This is unrelated to Quarto extension development, because you can work with `template.qmd` or `example.qmd` at the root of your extension to start testing/using it. (it does not even need to be under Git)
A lot of extensions do that, see https://m.canouil.dev/quarto-extensions/


https://github.com/user-attachments/assets/becac143-4c43-4b0e-abbb-868abb100dec
